### PR TITLE
api/database: Add the ability to delete layers

### PR DESCRIPTION
**Pull request by Quentin-M**

### CODE REVIEW

File: database/layer.go [hunk 1]
```diff
@@ -131,6 +131,59 @@ func InsertLayer(layer *Layer) error {
 	return nil
 }
 
+// DeleteLayer delete the specified layer and every layers that are based on that one, recursively.
```
Comment (jzelinskie) [inline database/layer.go]: DeleteLayer deletes the specified layer and any child layers that are dependent on the specified layer.

Also, comments in Go are 80 chars per line

Comment (Quentin-M) [inline database/layer.go]: Dang it. Such a pain in the butt to switch back and forth between 100C (Python) to 80C (Go).

Comment (jzelinskie) [inline database/layer.go]: Use [go-lint](https://github.com/golang/lint)

Comment (Quentin-M) [inline database/layer.go]: I do use it through https://atom.io/packages/go-plus, automatically on save but it does not wrap comments / enforce line length afaik.

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**